app/diagnostics.go

```diff
@@ -0,0 +1,394 @@
+// Copyright (c) 2015 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package app
+
+import (
+	"runtime"
+
+	"github.com/mattermost/platform/model"
+	"github.com/mattermost/platform/utils"
+	"github.com/segmentio/analytics-go"
+)
+
+const (
+	SEGMENT_KEY = "fwb7VPbFeQ7SKp3wHm1RzFUuXZudqVok"
+
+	TRACK_CONFIG_SERVICE      = "config_service"
+	TRACK_CONFIG_TEAM         = "config_team"
+	TRACK_CONFIG_SQL          = "config_sql"
+	TRACK_CONFIG_LOG          = "config_log"
+	TRACK_CONFIG_FILE         = "config_file"
+	TRACK_CONFIG_RATE         = "config_rate"
+	TRACK_CONFIG_EMAIL        = "config_email"
+	TRACK_CONFIG_PRIVACY      = "config_privacy"
+	TRACK_CONFIG_OAUTH        = "config_oauth"
+	TRACK_CONFIG_LDAP         = "config_ldap"
+	TRACK_CONFIG_COMPLIANCE   = "config_compliance"
+	TRACK_CONFIG_LOCALIZATION = "config_localization"
+	TRACK_CONFIG_SAML         = "config_saml"
+	TRACK_CONFIG_PASSWORD     = "config_password"
+	TRACK_CONFIG_CLUSTER      = "config_cluster"
+	TRACK_CONFIG_METRICS      = "config_metrics"
+	TRACK_CONFIG_WEBRTC       = "config_webrtc"
+	TRACK_CONFIG_SUPPORT      = "config_support"
+	TRACK_CONFIG_NATIVEAPP    = "config_nativeapp"
+	TRACK_CONFIG_ANALYTICS    = "config_analytics"
+
+	TRACK_ACTIVITY = "activity"
+	TRACK_LICENSE  = "license"
+	TRACK_SERVER   = "server"
+)
+
+var client *analytics.Client
+
+func SendDailyDiagnostics() {
+	if *utils.Cfg.LogSettings.EnableDiagnostics {
+		initDiagnostics()
+		trackActivity()
+		trackConfig()
+		trackLicense()
+		trackServer()
+	}
+}
+
+func initDiagnostics() {
+	if client == nil {
+		client = analytics.New(SEGMENT_KEY)
```

Comment:
I had a look at the segment-go source code and the answer is nothing. The events just go into a black hole. There are no errors thrown or returned or anything like that.